test/Microsoft.ML.StaticPipelineTesting/StaticPipeTests.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using FakeStaticPipes;
+using Microsoft.ML.Data.StaticPipe;
+using Microsoft.ML.Data.StaticPipe.Runtime;
+using Microsoft.ML.Runtime;
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.TestFramework;
+using System;
+using System.IO;
+using System.Text;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.ML.StaticPipelineTesting
+{
+
+    public abstract class MakeConsoleWork : IDisposable
+    {
+        private readonly ITestOutputHelper _output;
+        private readonly TextWriter _originalOut;
+        private readonly TextWriter _textWriter;
+
+        public MakeConsoleWork(ITestOutputHelper output)
+        {
+            _output = output;
+            _originalOut = Console.Out;
+            _textWriter = new StringWriter();
+            Console.SetOut(_textWriter);
+        }
+
+        public void Dispose()
+        {
+            _output.WriteLine(_textWriter.ToString());
+            Console.SetOut(_originalOut);
+        }
+    }
+
+    public sealed class StaticPipeTests : MakeConsoleWork
+    {
+        public StaticPipeTests(ITestOutputHelper output)
+            : base(output)
+        {
+        }
+
+        [Fact]
+        public void SimpleTextLoaderCopyColumnsTest()
+        {
+            var env = new TlcEnvironment(new SysRandom(0), verbose: true);
+
+            const string data = "0 hello 3.14159 -0 2\n"
+                + "1 1 2 4 15";
+            var dataSource = new BytesStreamSource(data);
+
+            var text = TextLoader.CreateReader(env, ctx => (
+                label: ctx.LoadBool(0),
+                text: ctx.LoadText(1),
+                numericFeatures: ctx.LoadFloat(2, null)), // If fit correctly, this ought to be equivalent to max of 4, that is, length of 3.
+                dataSource, separator: ' ');
+
+            // While we have a type-safe wrapper for `IDataView` it is utterly useless except as an input to the `Fit` functions
+            // of the other statically typed wrappers. We perhaps ought to make it useful in its own right, but perhaps not now.
+            // For now, just operate over the actual `IDataView`.
+            var textData = text.Read(dataSource).Wrapped;
+
+            var schema = textData.Schema;
+            // First verify that the columns are there. There ought to be at least one column corresponding to the identifiers in the tuple.
```

Comment:
>// [](start = 12, length = 2)

maybe put below code into a separate test method, like 'CheckSmth'?

Reply:
OK. And please tell me you're insisting on that name. :)

---
In reply to: [214165914](https://github.com/dotnet/machinelearning/pull/778#discussion_r214165914) [](ancestors = 214165914)